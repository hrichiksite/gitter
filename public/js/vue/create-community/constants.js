export const CREATE_COMMUNITY_STEP_MAIN = 'step-main';
export const CREATE_COMMUNITY_STEP_BACKING_ENTITY_GITLAB = 'step-backing-entity-gitlab';
export const CREATE_COMMUNITY_STEP_BACKING_ENTITY_GITHUB = 'step-backing-entity-github';

export const CREATE_COMMUNITY_ENTITY_TYPE_TAB_ORGS_STATE = 'tab-orgs';
export const CREATE_COMMUNITY_ENTITY_TYPE_TAB_REPOS_STATE = 'tab-repos';

export const slugAvailabilityStatusConstants = {
  PENDING: 'pending',
  AVAILABLE: 'available',
  GITHUB_CLASH: 'github-clash',
  AUTHENTICATION_FAILED: 'authentication-failed',
  UNAVAILABLE: 'unavailable',
  INVALID: 'invalid'
};
