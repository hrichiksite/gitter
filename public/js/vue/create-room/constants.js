export const roomAvailabilityStatusConstants = {
  PENDING: 'pending',
  UNAVAILABLE: 'unavailable',
  VALIDATION_FAILED: 'validation_failed',
  ILLEGAL_NAME: 'illegal_name',
  REPO_CONFLICT: 'repo_conflict',
  INSUFFICIENT_PERMISSIONS: 'insufficient_permissions',
  GROUP_REQUIRED: 'group_required'
};
