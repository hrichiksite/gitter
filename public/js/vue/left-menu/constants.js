export const LEFT_MENU_ALL_STATE = 'all';
export const LEFT_MENU_SEARCH_STATE = 'search';
export const LEFT_MENU_PEOPLE_STATE = 'people';
export const LEFT_MENU_CREATE_STATE = 'create';
export const LEFT_MENU_TOGGLE_STATE = 'toggle';
export const LEFT_MENU_ANNOUNCEMENTS_STATE = 'announcements';

export const VALID_LEFT_MENU_STATES = [
  LEFT_MENU_ALL_STATE,
  LEFT_MENU_SEARCH_STATE,
  LEFT_MENU_PEOPLE_STATE,
  LEFT_MENU_CREATE_STATE,
  LEFT_MENU_TOGGLE_STATE,
  LEFT_MENU_ANNOUNCEMENTS_STATE
];
