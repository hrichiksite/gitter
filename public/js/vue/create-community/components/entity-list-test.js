const mount = require('../../__test__/vuex-mount');
const {
  createOrgGitlabGroupFixture,
  createOrgGithubOrgFixture,
  createRepoGithubRepoFixture
} = require('../../__test__/fixture-helpers');
const { default: EntityList } = require('./entity-list.vue');

import { CREATE_COMMUNITY_STEP_MAIN } from '../constants';

describe('EntityList', () => {
  it('empty list matches snapshot', () => {
    const { wrapper } = mount(
      EntityList,
      {
        displayName: 'orgs',
        list: [],
        loading: false,
        error: null
      },
      store => {
        store.state.createCommunity.selectedBackingEntity = null;
      }
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('loading matches snapshot', () => {
    const { wrapper } = mount(
      EntityList,
      {
        displayName: 'orgs',
        list: [],
        loading: true,
        error: null
      },
      store => {
        store.state.createCommunity.selectedBackingEntity = null;
      }
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('error matches snapshot', () => {
    const { wrapper } = mount(
      EntityList,
      {
        displayName: 'orgs',
        list: [],
        loading: false,
        error: new Error('my fake error')
      },
      store => {
        store.state.createCommunity.selectedBackingEntity = null;
      }
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('GitLab group list matches snapshot', () => {
    const { wrapper } = mount(
      EntityList,
      {
        displayName: 'orgs',
        list: [
          createOrgGitlabGroupFixture('gitlab-org/gitter'),
          createOrgGitlabGroupFixture('gitlab-org')
        ],
        loading: false,
        error: null
      },
      store => {
        store.state.createCommunity.selectedBackingEntity = null;
      }
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('GitHub org list matches snapshot', () => {
    const { wrapper } = mount(
      EntityList,
      {
        displayName: 'orgs',
        list: [createOrgGithubOrgFixture('gitlabhq'), createOrgGithubOrgFixture('gitterHQ')],
        loading: false,
        error: null
      },
      store => {
        store.state.createCommunity.selectedBackingEntity = null;
      }
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('GitHub repo list matches snapshot', () => {
    const { wrapper } = mount(
      EntityList,
      {
        displayName: 'orgs',
        list: [
          createRepoGithubRepoFixture('gitterHQ/gitter'),
          createRepoGithubRepoFixture('gitlabhq/gitlabhq')
        ],
        loading: false,
        error: null
      },
      store => {
        store.state.createCommunity.selectedBackingEntity = null;
      }
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  describe('clicking item', () => {
    it('clicking entity selects the entity', () => {
      const org = createOrgGithubOrgFixture('gitterHQ');
      const { wrapper, stubbedActions } = mount(
        EntityList,
        {
          displayName: 'orgs',
          list: [org],
          loading: false,
          error: null
        },
        store => {
          store.state.createCommunity.selectedBackingEntity = null;
        }
      );

      wrapper.find('.entity-list-item-link').trigger('click');

      expect(stubbedActions.createCommunity.setSelectedBackingEntity).toHaveBeenCalledWith(
        expect.anything(),
        org,
        undefined
      );

      expect(stubbedActions.createCommunity.moveToStep).toHaveBeenCalledWith(
        expect.anything(),
        CREATE_COMMUNITY_STEP_MAIN,
        undefined
      );
    });

    it('clicking already selected entity deselects that entity', () => {
      const org = createOrgGithubOrgFixture('gitterHQ');
      const { wrapper, stubbedActions } = mount(
        EntityList,
        {
          displayName: 'orgs',
          list: [org],
          loading: false,
          error: null
        },
        store => {
          store.state.createCommunity.selectedBackingEntity = org;
        }
      );

      wrapper.find('.entity-list-item-link').trigger('click');

      expect(stubbedActions.createCommunity.setSelectedBackingEntity).toHaveBeenCalledWith(
        expect.anything(),
        null,
        undefined
      );

      expect(stubbedActions.createCommunity.moveToStep).toHaveBeenCalledWith(
        expect.anything(),
        CREATE_COMMUNITY_STEP_MAIN,
        undefined
      );
    });
  });
});
