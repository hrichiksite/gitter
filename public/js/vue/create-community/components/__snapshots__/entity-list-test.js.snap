// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityList GitHub org list matches snapshot 1`] = `
<ul
  aria-live="polite"
  class="entity-list"
>
  <!---->
   
  <li
    class="entity-list-item"
  >
    <a
      class="entity-list-item-link"
      data-disable-routing="1"
      href="https://github.com/gitlabhq"
    >
      <img
        class="entity-list-item-avatar"
        src="https://avatars2.githubusercontent.com/u/1086321?v=4"
      />
      
      gitlabhq
    
    </a>
  </li>
  <li
    class="entity-list-item"
  >
    <a
      class="entity-list-item-link"
      data-disable-routing="1"
      href="https://github.com/gitterHQ"
    >
      <img
        class="entity-list-item-avatar"
        src="https://avatars2.githubusercontent.com/u/1086321?v=4"
      />
      
      gitterHQ
    
    </a>
  </li>
</ul>
`;

exports[`EntityList GitHub repo list matches snapshot 1`] = `
<ul
  aria-live="polite"
  class="entity-list"
>
  <!---->
   
  <li
    class="entity-list-item"
  >
    <a
      class="entity-list-item-link"
      data-disable-routing="1"
      href="https://github.com/gitterHQ/gitter"
    >
      <img
        class="entity-list-item-avatar"
        src="https://avatars0.githubusercontent.com/u/5990364?v=4"
      />
      
      gitterHQ/gitter
    
    </a>
  </li>
  <li
    class="entity-list-item"
  >
    <a
      class="entity-list-item-link"
      data-disable-routing="1"
      href="https://github.com/gitlabhq/gitlabhq"
    >
      <img
        class="entity-list-item-avatar"
        src="https://avatars0.githubusercontent.com/u/5990364?v=4"
      />
      
      gitlabhq/gitlabhq
    
    </a>
  </li>
</ul>
`;

exports[`EntityList GitLab group list matches snapshot 1`] = `
<ul
  aria-live="polite"
  class="entity-list"
>
  <!---->
   
  <li
    class="entity-list-item"
  >
    <a
      class="entity-list-item-link"
      data-disable-routing="1"
      href="https://gitlab.com/groups/gitlab-org/gitter"
    >
      <img
        class="entity-list-item-avatar"
        src="https://assets.gitlab-static.net/uploads/-/system/group/avatar/1540914/icon_128x128.png"
      />
      
      gitlab-org/gitter
    
    </a>
  </li>
  <li
    class="entity-list-item"
  >
    <a
      class="entity-list-item-link"
      data-disable-routing="1"
      href="https://gitlab.com/groups/gitlab-org"
    >
      <img
        class="entity-list-item-avatar"
        src="https://assets.gitlab-static.net/uploads/-/system/group/avatar/1540914/icon_128x128.png"
      />
      
      gitlab-org
    
    </a>
  </li>
</ul>
`;

exports[`EntityList empty list matches snapshot 1`] = `
<ul
  aria-live="polite"
  class="entity-list"
>
  <li>
    No orgs available
  </li>
   
</ul>
`;

exports[`EntityList error matches snapshot 1`] = `
<ul
  aria-live="polite"
  class="entity-list"
>
  <li>
    <div
      class="error-text error-box"
    >
      Error: orgs can't be fetched.
    </div>
  </li>
   
</ul>
`;

exports[`EntityList loading matches snapshot 1`] = `
<ul
  aria-live="polite"
  class="entity-list"
>
  <li>
    Loading orgs... 
    <loading-spinner-stub />
  </li>
   
</ul>
`;
